packages/env/src/__tests__/environment.ts

```diff
@@ -114,6 +114,43 @@ describe("when env is set", () => {
       });
     });
   });
+
+  describe("with custom variables layer", () => {
+    let envPath: string;
+
+    const customVariablesLayer: VariablesLayer = (variables, componentName) => {
+      variables[componentName || "global"] = "overridden";
+      return variables;
+    };
+
+    beforeEach(() => {
+      env.env = "dev";
+      envPath = join(fixturePath, "environments", env.env);
+      env.addVariablesLayer(customVariablesLayer);
+    });
+
+    afterEach(() => {
+      env.removeVariablesLayer(customVariablesLayer);
+    })
+
+    describe("global", () => {
+      test("shold return global vars", () => {
+        expect(env.global()).toEqual({
+          ...require(envPath),
+          global: "overridden"
+        });
+      });
+    });
+
+    describe("component", () => {
+      test("shold return global + component vars", () => {
+        expect(env.component("foo")).toEqual({
+          ...merge(require(envPath), require(join(envPath, "foo"))),
+          foo: "overridden"
+        });
+      });
+    });
+  });
```

Comment:
You could add some test cases of the `removeVariablesLayer` method to make sure the layer is really removed.